rgdps/api/gd/user_relationships.py

```diff
@@ -0,0 +1,282 @@
+    result = await friend_requests.get_user(
+        ctx,
+        user.id,
+        page,
+        PAGE_SIZE,
+        is_sender_user_id=is_sender_user_id,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to view friend friend requests of {user.id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    if not result.requests:  # No requests found.
+        return responses.code(-2)
+
+    response = "|".join(
+        gd_obj.dumps(
+            [
+                gd_obj.create_friend_request(request.request),
+                gd_obj.create_friend_request_author_string(request.user),
+            ],
+        )
+        for request in result.requests
+    )
+    response += "#" + gd_obj.create_pagination_info(result.total, page, PAGE_SIZE)
+
+    logger.info(f"{user} successfully viewed friend requests.")
+    return response
+
+
+async def friend_request_upload(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    target_user_id: int = Form(..., alias="toAccountID"),
+    message: Base64String = Form("", alias="comment"),
+):
+
+    result = await friend_requests.create(
+        ctx,
+        user.id,
+        target_user_id,
+        message,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to upload friend request from {user.id} to {target_user_id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    logger.info(f"{user} successfully uploaded friend request to {target_user_id}.")
+    return responses.success()
+
+
+async def friend_request_read(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    request_id: int = Form(..., alias="requestID"),
+):
+
+    result = await friend_requests.mark_as_seen(
+        ctx,
+        user.id,
+        request_id,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to mark friend request {request_id} as seen with error {result!r}.",
+        )
+        return responses.fail()
+
+    logger.info(f"{user} successfully seen friend request {request_id}.")
+    return responses.success()
+
+
+async def friend_request_delete(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    target_id: int = Form(..., alias="targetAccountID"),
+    is_sender_user_id: bool = Form(0, alias="isSender"),
+    accounts: str | None = Form(None, alias="accounts"),
+):
+
+    if accounts:
+        accounts_list = [int(account) for account in accounts.split(",")]
+        await friend_requests.delete_multiple(
+            ctx,
+            user.id,
+            accounts_list,
+            is_sender_user_id=is_sender_user_id,
+        )
+
+        logger.info(
+            f"{user} successfully deleted friend request from/to {accounts_list}.",
+        )
+        return responses.success()
+
+    result = await friend_requests.delete(
```

Comment:
I would probably reuse `delete_multiple` feeding it only one id.

Reply:
Unless there is different logic for each.

Reply:
resolved in 006a05f